weather.py

```diff
@@ -7,6 +7,8 @@
 
 import numpy as np
 import pandas as pd
+from datetime import datetime
```

Comment:
Many of the attributes in the Weather class are pandas series with a datetime index.  It may not be necessary to import datetime again.